### ISSUE TITLE
test on the distance between end of trace and Station XIV is not present (or not active)

### ISSUE THREAD

**Issue by quchasz** [edk-software/edk-route-verifier#146]
Currently it is possible to approve Route with XIV Station far away from the end of trace.
It looks like regression to #59

**Comment by quchasz** [comment]
or it is the same issue as #149

**Comment by qooban** [comment]
Yes, it is regression to #59 . Unfortunatelly it is intentional regression as after running all routes from 2019 it appeared that the change introduced by #59 caused a lot of false positives verifications in case of circular routes (when Station XIV is close to route start). 

The reason: all the points are first mapped to the closest point on the route and it happens sometimes that Station XIV can be closer to the path start than to the end. Route length in such case is calculated improperly. 

Taking into account number of false positivies I must say that until we have clear requirements on setting stations in circular routes or routes with some parts shared, then I would not be able to implement it back.

**Comment by quchasz** [comment]
> UPDATE: To be clear - so right now in the latest implementation installed on production server - Station XIV placement is not taken into account in case of circular routes (routes where Station I and Station XIV are not further than in distance of 1km).

What do you mean by this? verifier classifies route as circle when Station 1 and Station 14 are located in distance up to 1km? If so this is incorrect assumption - imagine route like this one, when Station 1 is far away from the beginning of route:
<img width="641" alt="Screenshot 2020-03-04 at 12 54 53" src="https://user-images.githubusercontent.com/47346557/75877239-86613e00-5e17-11ea-816b-c625ce2bd7e9.png">


I opt to classify route as circle checking distance between first and last point on the trace line

**Comment by qooban** [comment]
@quchasz Sorry, my bad, double checked the code right now. The route is classified as circular by checking "between first and last point on the trace line" already. Still - I agree that it is not the perfect - but covers most of the cases properly.
